src/tasks/minions/agilityActivity.ts

```diff
@@ -90,8 +90,8 @@ export const agilityTask: MinionTask = {
 
 		// Calculate Crystal Shards for Priff
 		if (course.name === 'Prifddinas Rooftop Course') {
-			// 15 Shards per hour
-			loot.add('Crystal shard', Math.floor((duration / Time.Hour) * 15));
+			// 1 shard per lap
+			loot.add('Crystal shard', quantity);
```

Comment:
why is this 1 guaranteed shard per lap?

Reply:
That's what the update did, it was 1/3 before, now it's guaranteed, the 15/hr was just an estimate. Link is in the issue linked to this PR.